src/demo/providers/LogoutProvider.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { createContext, useEffect, useState } from "react";
+import { api, ExternalUser, SdkError, SdkErrors } from "@tonomy/tonomy-id-sdk";
+import useErrorStore from "../../common/stores/errorStore";
+
+interface LogoutContextProps {
+  logout: () => void;
+}
+
+interface LogoutProviderProps {
+  children: React.ReactNode;
+}
+
+export const LogoutContext = createContext<LogoutContextProps>({
+  logout: () => {},
+});
+
+const LogoutProvider: React.FC<LogoutProviderProps> = ({ children }) => {
+  const errorStore = useErrorStore();
+  const [user, setUser] = useState<ExternalUser | null>(null);
+
+  async function onRender() {
+    try {
+      const user = await api.ExternalUser.getUser();
+
+      setUser(user);
+    } catch (e) {
+      if (
+        e instanceof SdkError &&
+        (e.code === SdkErrors.AccountNotFound ||
+          e.code === SdkErrors.AccountDoesntExist ||
+          e.code === SdkErrors.UserNotLoggedIn)
+      ) {
+        // User not logged in
+        window.location.href = "/";
```

Comment:
this doesn't seem right. navigation between logged in and logged out should be done by state e.g.
https://github.com/remix-run/react-router/blob/dev/examples/auth/src/App.tsx#L87

Reply:
your code should also be able to resolve this noted issue
https://github.com/Tonomy-Foundation/Tonomy-App-Websites/blob/development/src/demo/pages/Callback.tsx#L23